Ready to build image caption data set.

oliverkinch opened the pull request:
image_caption_data_set.py indeholder class til at bygge image caption dataset.

build_image_caption_dataset.py er scriptet der skal køres, for at bygge image caption dataset.

Indeholder også minor ændringer til summary_data_set.py i hensyn til ændringer lavet i constants.py og utils.py, men det burde ikke være noget, som kræver det store code review.

oliverkinch committed:
Ready to build image caption data set.

image_caption_data_set.py indeholder class til at bygge image caption
dataset.

build_image_caption_dataset.py er scriptet der skal køres, for at bygge
image caption dataset.

Indeholder også minor ændringer til summary_data_set.py i hensyn til
ændringer lavet i constants.py og utils.py, men det burde ikke være
noget, som kræver det store code review.

oliverkinch commented:
Hvorfor fejler "black" i nedenstående tests? 🧐 

Lokalt er black vældig glad.
<img width="460" alt="image" src="https://user-images.githubusercontent.com/71556498/219676074-25e1ab73-71dd-405d-9985-5a7f3767d1a6.png">

Det må blive efter weekenden, at jeg opklarer det mysterie.

saattrupdan commented on src/nordjylland_news/constants.py:
```diff
@@ -1,17 +1,29 @@
 """Constants."""
 
+ARTICLES_API_URL = "https://public.nord.bazo.dk/v1/articles"
+
 # Max articles per page
 MAX_PER_PAGE = 100
 
+# Data set names
+SUMMARY_DATA_SET = "summary"
+IMAGE_CAPTION_DATA_SET = "image_caption"
+
 # Path to raw data
-RAW_DATA_PATH = "data/raw/data_set.jsonl"
+RAW_DATA_PATH = {
+    SUMMARY_DATA_SET: "data/raw/summary.jsonl",
+    IMAGE_CAPTION_DATA_SET: "data/raw/image_caption.jsonl",
+}
 
 # https://developer.bazo.dk/#876ab6f9-e057-43e3-897a-1563de34397e
 HEADERS = {"Accept": "application/json", "Authorization": "centered"}
 
 # Status codes
 STATUS_CODE_OK = 200
 TOO_MANY_REQUESTS = 429
+ERROR_500 = 500
 
 # Sleep time
 SLEEP_5_SECONDS = 5
+SLEEP_60_SECONDS = 60
+SLEEP_15_MINUTES = 900
```
De fleste af disse constants passer nok bedre i en config fil. Vi kan lige snakke om det når du er her.

saattrupdan commented on src/nordjylland_news/utils.py:
```diff
@@ -69,3 +82,64 @@ def html_to_text(html: str) -> str:
     return text
+
+
+def send_request(url):
```
```suggestion
def send_request(url) -> dict:
```

saattrupdan commented on src/nordjylland_news/utils.py:
```diff
@@ -69,3 +82,64 @@ def html_to_text(html: str) -> str:
+
+
+def get_page_with_articles_data(page: int) -> List[dict]:
+    """
```
Mangler beskrivelse.

saattrupdan reviewed (changes requested):
Ser supergodt ud! Jeg har kun et par få kommentarer. Kommentaren omkring configs kan vi lige tage sammen.

AJDERS commented:
@oliverkinch , @saattrupdan har travlt, så jeg reviewer osv. indtil han har mindre travlt :) Du kan fange mig til en snak om config, tests, og logging når det passer dig :)

AJDERS commented:
Jeg er ikke sikker på hvordan det er sket, men linting checket fejler. Du kan køre: `black .`, som burde fikse det :)

oliverkinch committed:
Fix changes requested

oliverkinch committed:
Update version for black - "black src/"

oliverkinch committed:
Substitute prints with logger.info()

oliverkinch committed:
Constants to config

oliverkinch committed:
Hydra and refactor using base class

File name changes because of inconsistent use of dataset/data_set

oliverkinch commented:
En lidt stor omgang. Jeg kan eventuelt lige sidde med, når du kigger det igennem? @AJDERS

AJDERS commented:
> En lidt stor omgang. Jeg kan eventuelt lige sidde med, når du kigger det igennem? @AJDERS

Yup 100%, lad os gøre det fredag. Vil du lige markere om rettelserne ovenfor er resolved eller ej?

oliverkinch commented:
> > En lidt stor omgang. Jeg kan eventuelt lige sidde med, når du kigger det igennem? @AJDERS
> 
> Yup 100%, lad os gøre det fredag. Vil du lige markere om rettelserne ovenfor er resolved eller ej?

Jeg har ligget syg i dag, så tror jeg bør holde mig fra kontoret i morgen. Kan vi tage den mandag?

oliverkinch committed:
Doc string til __init__

oliverkinch committed:
Mere docstring til __init__ i de tre classes

Docstring i base class og de to child classes er næsten identiske.
Kan man fx for child classes kun liste attributes som bliver
defineret i child class, og altså dermed ikke liste alle attributes
fra base class igen? Altså, blot nøjes med at referere til
base class, for at se base class attributes?

oliverkinch commented on src/nordjylland_news/base_dataset_class.py:
```diff
@@ -0,0 +1,134 @@
+        self.cfg = cfg
+        self.read_cfg()
+
+        self.logger = logging.getLogger(__name__)
```
Skift __name__ til self.dataset_name

oliverkinch commented on src/nordjylland_news/constants.py:
```diff
@@ -23,7 +8,7 @@
-SLEEP_5_SECONDS = 5
-SLEEP_60_SECONDS = 60
-SLEEP_15_MINUTES = 900
+# Sleep times
+SLEEP_SHORT = 5  # 5 seconds
+SLEEP_MEDIUM = 60  # 1 minute
+SLEEP_LONG = 900  # 15 minutes
```
Smid disse i config.yaml

oliverkinch committed:
Sleeps from constants.py to config.yaml

oliverkinch committed:
Read sleeps from config instead of constants.py

send_request er nu en method i base class i stedet for function i
utils.py.

oliverkinch committed:
Underscores instead of hyphens in cfg

oliverkinch commented:
Så skulle ændringerne, som vi (@saattrupdan) snakkede om tidligere, være ordnet :)

saattrupdan commented on src/nordjylland_news/image_caption_dataset.py:
```diff
@@ -0,0 +1,133 @@
+                New data to append to dataset.
+        """
+        # Initialize base class
+        super().__init__(dataset_name=dataset_name, cfg=cfg)
```
Dette sker helt automatisk, da argumenterne til baseklassen er de samme. Så du kan sagtens fjerne hele denne her init metode 🙂

saattrupdan reviewed (changes requested):
Meget lille ændring, og ellers ser det helt fint ud!

oliverkinch committed:
Class docstrings

saattrupdan reviewed (approved):
LGTM! 🎉
